Add broker note, refine some wording around network connection (#1787)
semgrep/semgrep-docs: docs/deployment/deployment-checklist.md

<IpAddresses />

> You may instead use the [Semgrep Network Broker](/docs/semgrep-ci/network-broker), which provides an alternative to allowlisting these IP addresses and has separate requirements.

Reviewer: ```suggestion
Alternatively, you can use the [Semgrep Network Broker](/docs/semgrep-ci/network-broker) to facilitate secure access with Semgrep instead of allowlisting these IP addresses.
```

#### Features that require inbound network connectivity

> - [Source code management connections](/docs/deployment/connect-scm#connect-to-on-premise-orgs-and-projects)

Reviewer: ```suggestion
- [Source code manager connections](/docs/deployment/connect-scm#connect-to-on-premise-orgs-and-projects)
```
